### ISSUE TITLE
BestModelCallback: _best_ema resets on resume, silently overwrites checkpoint_best_ema.pth

### ISSUE THREAD

**Issue by JuMa555** [roboflow/rf-detr#969]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

`BestModelCallback._best_ema` is a plain instance attribute initialized to `0.0` in `__init__`. The class does not override `state_dict()` / `load_state_dict()`, so `_best_ema` is never persisted into Lightning checkpoints. On resume via `trainer.fit(ckpt_path=...)`, the callback is reconstructed fresh, `_best_ema` resets to `0.0`, and the first post-resume validation trivially satisfies `ema_val > self._best_ema` — overwriting `checkpoint_best_ema.pth` with inferior weights.

The regular best checkpoint is unaffected because `best_model_score` is persisted by the parent `ModelCheckpoint`.

https://github.com/roboflow/rf-detr/blob/develop/src/rfdetr/training/callbacks/best_model.py

```python
self._best_ema: float = 0.0
```
```python
if ema_val > self._best_ema:
    self._best_ema = ema_val
```

`on_fit_end` is also affected — it uses `self._best_ema` for the regular vs EMA winner selection (`checkpoint_best_total.pth`), so the wrong source can be selected after a resume.

### Environment

RF-DETR: 1.6.3
OS: Linux
Python: 3.11
PyTorch: 2.6.0+cu124
GPU: NVIDIA GeForce RTX 3060

### Minimal Reproducible Example

1. Train with EMA enabled for enough epochs to establish a meaningful EMA best:

```python
from rfdetr import RFDETRLarge
model = RFDETRLarge()
model.train(
    dataset_dir="path/to/dataset",
    epochs=50,
    output_dir="output/",
    checkpoint_interval=5,
)
```

2. Interrupt training.
3. Resume:

```python
model = RFDETRLarge()
model.train(
    dataset_dir="path/to/dataset",
    epochs=50,
    output_dir="output/",
    checkpoint_interval=5,
    resume="output/latest.ckpt",
)
```

4. First post-resume validation logs `"Best EMA mAP improved to X.XXXX (epoch Y)"` with a value **lower** than the true pre-resume best.
5. `checkpoint_best_ema.pth` is silently overwritten.

Confirm by inspecting the saved checkpoint:

```python
ckpt = torch.load("output/checkpoint_best_ema.pth", map_location="cpu", weights_only=False)
print("Epoch:", ckpt["epoch"])  # Post-resume epoch, not the true-best epoch
```

### Additional finding: does `checkpoint_best_regular.pth` save EMA weights instead of regular weights?

`_save_checkpoint` (line 162–163) checks `self._monitor_ema is not None` and, when true, saves EMA weights via `_get_ema_model_state_dict` — even though the file is named `checkpoint_best_regular.pth` and the documentation describes it as "best checkpoint based on validation score, using the raw (non-EMA) model weights".

It seems that, when EMA is active:

- `checkpoint_best_regular.pth` is triggered by the regular metric (`val/mAP_50_95`) but contains EMA weights
- `checkpoint_best_ema.pth` is triggered by the EMA metric (`val/ema_mAP_50_95`) and also contains EMA weights
- No checkpoint preserves the actual regular (non-EMA) model weights

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!
